spec/requests/users_controller_spec.rb

```diff
@@ -0,0 +1,24 @@
+# spec/requests/users_controller_spec.rb
+require 'rails_helper'
+
+RSpec.describe UsersController, type: :request do
+  describe 'GET #index' do
+    it 'returns a successful response' do
+      get users_path
+      expect(response).to have_http_status(:success)
+      expect(response).to render_template(:index)
+      expect(response.body).to include('users Section')
+    end
+  end
+
+  describe 'GET #show' do
+    let(:user) { User.create(name: 'Example User', photo: 'example.jpg', bio: 'Example Bio', posts_counter: 0) }
+
+    it 'returns a successful response' do
+      get user_path(user)
+      expect(response).to have_http_status(:success)
+      expect(response).to render_template(:show)
+      expect(response.body).to include('Selected user from the list.')
+    end
```

Comment:
- [ **OPTIONAL** ] What do you thing if you put each specific test in a `it` method for best practices and to make you code be more readable. **I recommend you apply this example in each Action of each file**(i.e. `users_controller_spec` and `posts_controller_spec`)
```ruby
# Example
    it 'returns a successful response' do
      expect(response).to have_http_status(:success)
    end

    it 'renders a good template' do
      expect(response).to render_template(:show)
    end
  
    it 'checks if body includes the correct placeholder text' do
      expect(response.body).to include('Selected user from the list.')
    end
```